api/v1/db/awsS3/aws-operations.js

```diff
@@ -52,18 +52,18 @@ const validateAwsS3 = async () => {
  * @param {string} bucket The bucket where the key will be searched
  * @returns {Promise} Promise object represents a boolean indicating if the key exists or not
  */
-const objectExists = (key, bucket = thisBucket) => new Promise((resolve, reject) => {
+const objectExists = async (key, bucket = thisBucket) => {
   const params = { Bucket: bucket, Key: key };
-  s3.headObject(params).promise().then(() => {
-    resolve(true);
-  }).catch((err) => {
+  try {
+    await s3.headObject(params).promise();
+    return true;
+  } catch (err) {
     if (err.code === 'NotFound') {
-      resolve(false);
-    } else {
-      reject(err);
+      return false;
     }
-  });
-});
+    throw err;
+  }
+};
```

Comment:
`objectExists` and `bucketExists` are almost the same, do you think we can combine them together?